### ISSUE TITLE
Cannot create new alert conditions

### ISSUE THREAD

**Issue by lennartkoopmann** [Graylog2/graylog2-web-interface#1042]
Getting a HTTP 500 when trying to create a new alert condition for a stream. I am on the latest 1.0 branch so I guess this was also released in beta.2.

````
2015-01-16 21:06:24,863 ERROR: org.graylog2.plugin.rest.AnyExceptionClassMapper - Unhandled exception in REST resource
java.lang.IllegalArgumentException: The template variable 'conditionId' has no value
	at org.glassfish.jersey.uri.UriTemplate$1ValuesFromArrayStrategy.valueFor(UriTemplate.java:1017)
	at org.glassfish.jersey.uri.UriTemplate.resolveTemplate(UriTemplate.java:706)
	at org.glassfish.jersey.uri.UriTemplate.createUriComponent(UriTemplate.java:1027)
	at org.glassfish.jersey.uri.UriTemplate.createURIWithStringValues(UriTemplate.java:961)
	at org.glassfish.jersey.uri.UriTemplate.createURIWithStringValues(UriTemplate.java:906)
	at org.glassfish.jersey.uri.UriTemplate.createURI(UriTemplate.java:871)
	at org.glassfish.jersey.uri.internal.JerseyUriBuilder._build(JerseyUriBuilder.java:906)
	at org.glassfish.jersey.uri.internal.JerseyUriBuilder.build(JerseyUriBuilder.java:823)
	at org.graylog2.rest.resources.streams.alerts.StreamAlertConditionResource.create(StreamAlertConditionResource.java:102)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:151)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:172)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:152)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:104)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:384)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:342)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:101)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:297)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:254)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1030)
	at org.graylog2.jersey.container.netty.NettyContainer.messageReceived(NettyContainer.java:356)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.handler.execution.ChannelUpstreamEventRunnable.doRun(ChannelUpstreamEventRunnable.java:43)
	at org.jboss.netty.handler.execution.ChannelEventRunnable.run(ChannelEventRunnable.java:67)
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176)
	at org.jboss.netty.handler.execution.MemoryAwareThreadPoolExecutor$MemoryAwareRunnable.run(MemoryAwareThreadPoolExecutor.java:622)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
````

**Comment by lennartkoopmann** [comment]
The condition is successfully created though. It is listed in the list of conditions and alerts are fired.

**Comment by kroepke** [comment]
Bernd was already on this earlier today.

**Comment by bernd** [comment]
Yes, I fixed the problems already in server.

**Comment by bernd** [comment]
Duplicate of Graylog2/graylog2-server#891 and fixed in Graylog2/graylog2-server@acacb43 and Graylog2/graylog2-server@536a8fa.
